### PR TITLE
SuggestCommand: sort suggested modules based on faculty modules and predetermined importance

### PR DESCRIPTION
**Pull request by rongjiecomputer**
After the prereq/preclu filtering, suggested modules will be sorted with stable sort with these criteria:
- Sort modules from a faculty/major (e.g. `CS`, `IS`, `MA` first before anything else).
- Sort modules based on a hard-coded list.

Sorting with `FocusArea` is not yet implemented.

Drive-by-fix:
- Remove dead test files for `ModuleInfoStorage` (no longer exists).
- Some code reformatting.

### CODE REVIEW

File: src/main/java/seedu/planner/model/ModulePlanner.java [hunk 1]
```diff
@@ -305,11 +321,57 @@ public void suggestModules(int index) {
...
+            logger.warning("Unable to read majorDescription file. Start with an empty map.");
+            map = new HashMap<>();
+        }
+
+        // Note: Collections.sort uses stable sort when sorting objects, which we are exploiting here so that
+        // we can chain our sorting and still making sure that the order created by each comparator is preserved.
+        //
+        // The order of comparators you applied to the list matters!
+
+        if (map.containsKey(userProfile.getMajor())) {
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/ModulePlanner.java]: Might want to move some parts into their own methods

File: src/main/java/seedu/planner/model/util/ModuleUtil.java [hunk 2]
```diff
@@ -99,6 +100,66 @@ public static boolean isModuleAvailableToTake(List<Module> modulesTaken, List<Mo
...
+        return rank;
+    }
+
+    /**
+     * Give {@code moduleCode} a ranking based on its position in {@code priorityList}.
+     */
+    public static int rankModuleCodeFromPriorityList(String moduleCode, List<ModuleDescription> priorityList) {
+        int rank = 0;
+        for (ModuleDescription m : priorityList) {
+            if (m.getName().equals(moduleCode)) {
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/util/ModuleUtil.java]: I think `getCode()` would be a better method name. Cos we're comparing between two codes

File: src/test/java/seedu/planner/model/course/MajorDescriptionTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        modules.add(new ModuleDescription("CS2106", ProgrammeRequirement.FOUNDATION));
+        modules.add(new ModuleDescription("CS3230", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3880", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3200", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3202", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("IS4010", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3200", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+        modules.add(new ModuleDescription("CP3107", ProgrammeRequirement.INDUSTRIAL_EXPERIENCE_REQUIREMENT));
+
+        modules.add(new ModuleDescription("IS1103", ProgrammeRequirement.IT_PROFESSIONALISM));
```
Comment (GabrielYik) [inline src/test/java/seedu/planner/model/course/MajorDescriptionTest.java]: Perhaps can move this line up to be consistent with the rest. Or we can separate the lines based on their module type

File: src/test/java/seedu/planner/model/course/MajorDescriptionTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        // same object -> returns true
+        assertTrue(cs.equals(cs));
+
+        // null -> return false
+        assertFalse(cs.equals(null));
+
+        // different type -> returns false
+        assertFalse(cs.equals(5));
+
+        ModuleDescription cs1010_fake = new ModuleDescription("CS1010", ProgrammeRequirement.IT_PROFESSIONALISM);
```
Comment (GabrielYik) [inline src/test/java/seedu/planner/model/course/MajorDescriptionTest.java]: Should be camelCase to following coding standard

File: src/test/java/seedu/planner/model/course/ModuleDescriptionTest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+        // different type -> returns false
+        assertFalse(cs1010.equals(5));
+
+        // different module code -> returns false
+        ModuleDescription cs1231 = new ModuleDescription("CS1231", ProgrammeRequirement.FOUNDATION);
+        assertFalse(cs1010.equals(cs1231));
+
+        // different programme requirement -> returns false
+        ModuleDescription cs1010_fake = new ModuleDescription("CS1010", ProgrammeRequirement.IT_PROFESSIONALISM);
```
Comment (GabrielYik) [inline src/test/java/seedu/planner/model/course/ModuleDescriptionTest.java]: camelCase too

File: src/main/java/seedu/planner/model/util/ModuleUtil.java (hunk 2 above)
Comment (rongjiecomputer) [reply src/main/java/seedu/planner/model/util/ModuleUtil.java]: Done.

File: src/test/java/seedu/planner/model/course/MajorDescriptionTest.java (hunk 3 above)
Comment (rongjiecomputer) [reply src/test/java/seedu/planner/model/course/MajorDescriptionTest.java]: Done. Moved the line up.

File: src/test/java/seedu/planner/model/course/MajorDescriptionTest.java (hunk 4 above)
Comment (rongjiecomputer) [reply src/test/java/seedu/planner/model/course/MajorDescriptionTest.java]: Done.

File: src/test/java/seedu/planner/model/course/ModuleDescriptionTest.java (hunk 5 above)
Comment (rongjiecomputer) [reply src/test/java/seedu/planner/model/course/ModuleDescriptionTest.java]: Done.

File: src/main/java/seedu/planner/model/ModulePlanner.java (hunk 1 above)
Comment (rongjiecomputer) [reply src/main/java/seedu/planner/model/ModulePlanner.java]: Done.

**Comment by GabrielYik** [comment]
I ok to merge. Can wait for Hilda to review

**Comment by rongjiecomputer** [comment]
@Hilda-Ang Can you take a look since both of us touch the code for `SuggestCommand`?

**Pull request merged**